pilot/cmd/pilot-agent/metrics/metrics.go

```diff
@@ -12,25 +12,37 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package status
+package metrics
 
-import "istio.io/pkg/monitoring"
+import (
+	"time"
+
+	"istio.io/pkg/monitoring"
+)
 
 var (
 	typeTag = monitoring.MustCreateLabel("type")
 
+	// StartupTime measures the time it takes for the agent to get ready Note: This
+	// is dependant on readiness probes. This means our granularity is correlated to
+	// the probing interval.
+	startupTime = monitoring.NewGauge(
+		"startup_time",
+		"The time from the process starting to being marked ready.",
+	)
```

Comment:
curious: do you want to track this by proxy version or anything similar?

Reply:
How is this normally done? It seems like *any* metric we might want to.. but is it something that should be done by prom or higher level tool? Not sure if it has a way to "join" the pod name or other label and fetch the label

Reply:
the only reason i asked is that this metric is proxy-specific -- and the agent is possibly able to know the version at startup time (and could record it directly possibly).